src/MakersOfDenmark.WebAPI/Controllers/AdminMakerSpaceController.cs

```diff
@@ -62,7 +62,7 @@ public async Task<IActionResult> EditMakerSpaceOrganization(EditMakerSpaceOrgani
         public async Task<IActionResult> AddMakerSpaceTools(AddMakerSpaceTool request)
         {
             await _mediator.Send(request);
-            return CreatedAtAction(nameof(MakerSpaceController.GetTools), new { Id = request.MakerSpaceId}, request);
+            return CreatedAtAction("GetTools", "MakerSpace", new { makerSpaceId = request.MakerSpaceId }, request);
```

Comment:
Jeg tror det er smartest at bruge `nameof(GetTools), nameof(MakerSpaceController)....` i stedet for streng værdierne

Reply:
Din ændring gør det samme som min oprindelige statement. Har du nogen holdning til at ændre den?
Den oprindelige sætning er mere holdbar hvis man laver nogle refactorings senere.

Reply:
Jeg kunne ikke få koden tidligere til at virke. Den gemte fint i databasen, men responsed var ikke med location header men derimod en exception. Når man bruger CreatedAtAction(nameof(MakerSpaceController.GetTools) bliver det til strengen GetTools. 

Jeg tror at den pr. default kigger i den controller den allerede er i, derfor har du muligheden for at specificere en controller routen skal kigge i. 

Jeg ved så ikke hvorfor, men når du har specificeret en controller med nameof(MakerSpaceController) så kan den hellere ikke finde ud af, at route korrekt. Det skal derfor være MakerSpace man indsætter, fordi en nameof(MakerSpaceController) er det samme som MakerSpaceController. 

Jeg tror det er fordi den gemmer alle dine APIs routes internt i ASP.net som det er angivet i dit routing pattern, og så bruger de en helper metode til at finde ud af om routen eksisterer. Det defaulte pattern er jo api/makerspace/.... og ikke api/makerspacecontroller/...

Jeg får følgende fejl med denne implementation 
```csharp
return CreatedAtAction(nameof(MakerSpaceController.GetTools), new { makerSpaceId = request.MakerSpaceId }, request);
```

når jeg opretter et gyldigt tool.

```
System.InvalidOperationException: No route matches the supplied values.
   at Microsoft.AspNetCore.Mvc.CreatedAtActionResult.OnFormatting(ActionContext context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ObjectResultExecutor.ExecuteAsyncCore(ActionContext context, ObjectResult result, Type objectType, Object value)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ObjectResultExecutor.ExecuteAsync(ActionContext context, ObjectResult result)
   at Microsoft.AspNetCore.Mvc.ObjectResult.ExecuteResultAsync(ActionContext context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeResultAsync>g__Logged|21_0(ResourceInvoker invoker, IActionResult result)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResultFilterAsync>g__Awaited|29_0[TFilter,TFilterAsync](ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResultExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.ResultNext[TFilter,TFilterAsync](State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeResultFilters()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Logged|17_1(ResourceInvoker invoker)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)

HEADERS
=======
Accept: */*
Accept-Encoding: gzip, deflate, br
Accept-Language: da,en;q=0.9,en-GB;q=0.8,en-US;q=0.7
Connection: close
Content-Length: 84
Content-Type: application/json
Host: localhost:44302
Referer: https://localhost:44302/swagger/index.html
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.111 Safari/537.36 Edg/86.0.622.51
origin: https://localhost:44302
sec-fetch-site: same-origin
sec-fetch-mode: cors
sec-fetch-dest: empty
```